### PR TITLE
Fix flag parsing for hyperkube

### LINKED ISSUE #57628: local-up-cluster startup failed
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line: 
>
/kind bug
> /kind feature

**What happened**:
I pull the latest code, and run with below cmd. But the cluster is not up. 
```
root@SZX1000353068:/mnt/go/src/k8s.io/kubernetes# ./hack/local-up-cluster.sh
```

And I notice the kube-apiserver.log, 
```
unknown flag: --authorization-mode
Usage of /mnt/go/src/k8s.io/kubernetes/_output/local/bin/linux/amd64/hyperkube:
...
```
**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):  GitCommit:"980a5e80b12b35cb724aa276b8b589b5e96dd80f"
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by dims**
**What this PR does / why we need it**:
When #57132 was merged, we broke flag parsing for hyperkube

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #57628

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by dims** [comment]
cc @zjj2wry

**Comment by wanghaoran1988** [comment]
@dims The kube-scheduler still cannot start with err log:
```
I1227 19:33:56.663384   15243 server.go:552] Version: v1.10.0-alpha.1.281+c45449b4328cf1-dirty
Error: unsupported algorithm source: {<nil> <nil>}
```

**Comment by dims** [comment]
@wanghaoran1988 all the CI tests are green ... and i tried `hack/local-up-cluster.sh` as well. 

How are you trying to run kube-scheduler? what parameters?

**Comment by dims** [comment]
/close

In favor of #57650 from from @hzxuzhonghu

**Comment by wanghaoran1988** [comment]
@dims I tried `hack/local-up-cluster.sh` and the scheduler cannot startup

File: cmd/hyperkube/kube-scheduler.go [hunk 1]
```diff
@@ -17,38 +17,23 @@ limitations under the License.
...
+		},
 	}
-
-	serverFlags := hks.Flags()
-	serverFlags.AddFlagSet(command.Flags())
-
-	// FIXME this is here because hyperkube does its own flag parsing, and we need
-	// the command to know about the go flag set. Remove this once hyperkube is
-	// refactored to use cobra throughout.
-	command.Flags().AddGoFlagSet(flag.CommandLine)
```
[the remark this answers is not in the document]
Comment (nikhita) [reply cmd/hyperkube/kube-scheduler.go]: If hyperkube does use Cobra throughout, it should fix https://github.com/kubernetes/kubernetes/issues/12142.

Ref https://github.com/kubernetes/kubernetes/pull/44488#issuecomment-303099276 and https://github.com/kubernetes/kubernetes/issues/34732.

**Pull request closed without merging**